src/bindings/wasm/tvgWasmLottieAnimation.cpp

```diff
@@ -425,6 +456,10 @@ class __attribute__((visibility("default"))) TvgLottieAnimation
 
 EMSCRIPTEN_BINDINGS(thorvg_bindings)
 {
+    #ifdef THORVG_WG_RASTER_SUPPORT
+        emscripten::function("initWgInstance", &initWgInstance);
+    #endif
```

Comment:
@hermet I know avoiding helper function like this would be better. But without this interface, it is hard to make sure WGPU initialization called once in asynchronous and multiple players creation.

Furthermore if constructor has WGPU request part, emscripten forcibly makes constructor as a asynchronous function.

Then We have to call like this:
```ts
this._TVG = await (new _module.TvgLottieAnimation(engine, `#${this._canvas!.id}`));
```

Using `await` keyword on constructor is sort of unusual JavaScript constructor, it will make users confused.

Thus I introduced new helper function, this interface can provide APIs with better usage of WGPU initialization.

Reply:
@tinyjin If so, ThorVG may need to redesign the consistent engine initialization interfaces. Could you please suggest an example JS usage for `initWgInstance`?

Reply:
Somehow, this kind of usage might look better and be preferred.

```
   await _module.init();  //hide wg instances get ready in it.
   this._TVG = new _module.TvgLottieAnimation(engine, `#${this._canvas!.id}`);
```

```
    this._TVG = new _module.TvgLottieAnimation(engine, `#${this._canvas!.id}`);
    await _module.ready();  //hide wg instances get ready in it.
    ...
```

Reply:
@hermet

I tried to move single initialization status to C++ binding side. To do that, we have to call `emscripten_sleep` function for asynchronous logic. However duplicated calls of `emscripten_sleep` won't be properly worked.

For example, when we have multiple players, we need condition to block for duplicated WG initialization logics in C++ side, then multiple `emscripten_sleep()` are definitely called at the same time.

But that point makes WG initialization fall into infinite loop, because C++ doesn't actually support asynchronous processing in single thread, conversely JavaScript does even in single thread.

While C++ waits for requesting WG adapter/device by calling `emscripten_sleep`, another `emscripten_sleep()` from the other players(to avoid duplicated calls) raises deadlock and they just keep waiting without running back for prior WG initialization. (_Meaning that we'll meet infinite loop in multiple players case_).

Thus in my opinion, **we should have condition for avoiding multiple WG initializations in JS side not binding.**

Reply:
@tinyjin module.init() can be dealt with a global resource, it's expected to call just once in app, not multiple times.  What do you think?
```
   //must be called just once.
   await _module.init();  //hide wg instances get ready in it.
   
   ...
   //per player instances.
   this._TVG = new _module.TvgLottieAnimation(engine, `#${this._canvas!.id}`);  
   
```

Reply:
@hermet It's possible, but when module loaded once in global, we don't know WG instance is required.
Some users could just import SW+WG combined binary and decide not to use WG. Then preloaded WG Instances will be occupied some resources.

If it's not critical, global initialization in out of component would be good.

Reply:
@tinyjin Please take a look at the engine initialization of ThorVG. Ultimately, likewise, the user will decide the engine based on their preferences and requirements. The module init could have an engine types and initialize only with them.
```
await _module.init("default");  
await _module.init("software"); 
await _module.init("webgpu");
await _module.init("all"); //all possible engines
```

Please weigh the pros and cons of this approach compared to your suggestion. Which one is more difficult to understand and use from the WASM interfaces?

Reply:
I agree with this way. Module initialization is lazy, some users might want to have their own strategy. In the end, we should let users choose options.

Currently, we don't have external interfaces other than component tag. I assume this update will come along with out of this patch.

So far, we can decide from two options:
1. Multiple attempt, blocking duplicated call from JS side
2. Global one-time initialization when module loaded

Option 1 is good for users. They won't waste resources. WG resources are only loaded when user has WG based animation.

Option 2 is good for us. Even if it could bring redundant resources, we can keep consistent and simpler interface.

If I had to decide, I'd go with Option 1 because user side benefit is priority, I believe.

Reply:
> some users might want to have their own strategy

Ok, could you please suggest this you are considering?

> Currently, we don't have external interfaces other than component tag. I assume this update will come along with out of this patch.

May I ask what is `component tag` exactly? If it's necessary it's time to design the interface.

> Option 2 is good for us. Even if it could bring redundant resources, we can keep consistent and simpler interface.

@tinyjin I didn't suggest for us, but for developers who will implement directly using the WASM binding APIs. Now it's time to ensure there were 'redundant resources.' If it does, could you provide any measurable data?